lib/hcrypto/rsa-ltm.c

```diff
@@ -517,6 +517,7 @@ ltm_rsa_generate_key(RSA *rsa, int bits, BIGNUM *e, BN_GENCB *cb)
 {
     mp_int el, p, q, n, d, dmp1, dmq1, iqmp, t1, t2, t3;
     int counter, ret, bitsp;
+    int trials = mp_prime_rabin_miller_trials(bits);
```

Comment:
This shouldn't be fixed to the total number of bits!

You're generating two primes with roughly `bits/2` and as the number of trials decreases with a growing number of bits this wouldn't be an appropriate number to check the two half-sized primes.

Reply:
I'm out of my depth here so – I can either apply your patches (and get @vdukhovni to review) or revert to Love's original `mp_find_prime()` and bring it direct into libhcrypto.

Reply:
OK, got it, We need to increase the number of trials the shorter the number of bits. Patch incoming.